resilience4j-micrometer/src/main/java/io/github/resilience4j/micrometer/Timer.java

```diff
@@ -0,0 +1,428 @@
+     */
+    static Timer of(String name, MeterRegistry registry, TimerConfig timerConfig) {
+        return of(name, registry, timerConfig, emptyMap());
+    }
+
+    /**
+     * Creates a Timer with a custom Timer configuration.
+     *
+     * @param name        the name of the Timer
+     * @param registry    the registry to bind Timer to
+     * @param timerConfig a custom Timer configuration
+     * @param tags        tags to assign to the Timer
+     * @return a Timer with a custom Timer configuration.
+     */
+    static Timer of(String name, MeterRegistry registry, TimerConfig timerConfig, Map<String, String> tags) {
+        return new TimerImpl(name, registry, timerConfig, tags);
+    }
+
+    /**
+     * Decorates CompletionStageSupplier with Timer
+     *
+     * @param timer    the timer
+     * @param supplier completion stage supplier
+     * @param <T>      type of completion stage result
+     * @return decorated supplier
+     */
+    static <T> Supplier<CompletionStage<T>> decorateCompletionStage(Timer timer, Supplier<CompletionStage<T>> supplier) {
+        return () -> {
+            CompletableFuture<T> promise = new CompletableFuture<>();
+            Context context = timer.createContext();
+            try {
+                supplier.get().whenComplete((output, throwable) -> {
+                    if (throwable != null) {
+                        context.onFailure(throwable);
+                        promise.completeExceptionally(throwable);
+                    } else {
+                        context.onSuccess(output);
+                        promise.complete(output);
+                    }
+                });
+            } catch (Exception e) {
+                context.onFailure(e);
+                promise.completeExceptionally(e);
+            }
+            return promise;
+        };
+    }
+
+    /**
+     * Decorates CheckedSupplier with Timer
+     *
+     * @param timer    the timer
+     * @param supplier the supplier
+     * @param <T>      the supplier result
+     * @return decorated supplier
+     */
+    static <T> CheckedSupplier<T> decorateCheckedSupplier(Timer timer, CheckedSupplier<T> supplier) {
+        return () -> {
+            Context context = timer.createContext();
+            try {
+                T output = supplier.get();
+                context.onSuccess(output);
+                return output;
+            } catch (Exception e) {
+                context.onFailure(e);
+                throw e;
+            }
+        };
+    }
+
+    /**
+     * Decorates CheckedRunnable with Timer
+     *
+     * @param timer    the timer
+     * @param runnable the original runnable
+     * @return decorated runnable
+     */
+    static CheckedRunnable decorateCheckedRunnable(Timer timer, CheckedRunnable runnable) {
+        return () -> {
+            Context context = timer.createContext();
+            try {
+                runnable.run();
+                context.onSuccess(null);
```

Comment:
Let's change `context.onSuccess(null)` to `context.onSuccess()` to make it analog to CircuitBreaker or Retry.
In that chase the SUCCESS_RESULT_NAME_RESOLVER must not be able to handle null values.

Reply:
Instead of `onSuccess()` I added a `onNoResult()` method to indicate that this behaviour fires when the decorated operation returns nothing (is void). I can think of an alternative name like `onVoid()`. Which name do you prefer?

Reply:
IMO "on success" refers to both, "on result" and "on no result".

Reply:
I prefer onSuccess(), because it is similar to all other implementations

Reply:
Done